ecs_helper/cmd/pact_tags/main.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"os"
+	"fmt"
+	"net/http"
+	"encoding/base64"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+)
+
+func basicAuth(username, password string) string {
+	auth := username + ":" + password
+	return base64.StdEncoding.EncodeToString([]byte(auth))
+}
+
+func getPactPassword(roletoassume string, pactsecretkey string) string {
+	mysession := session.Must(session.NewSession())
+	creds := stscreds.NewCredentials(mysession, roletoassume)
+	cfg := aws.Config{Credentials: creds,Region: aws.String("eu-west-1")}
+	sess := session.Must(session.NewSession(&cfg))
+	sm := secretsmanager.New(sess)
+	secretkey := pactsecretkey
+	pact_password, err := sm.GetSecretValue(&secretsmanager.GetSecretValueInput{SecretId: &secretkey})
+	if err != nil {
+		panic(err.Error())
+	}
+	return *pact_password.SecretString
+}
+
+func main() {
+
+	base_url := os.Getenv("PACT_BROKER_BASE_URL")
+	consumer_version := os.Getenv("PACT_CONSUMER_VERSION")
+	api_version := os.Getenv("PACT_API_VERSION")
+
+	if len(base_url) == 0 || len(consumer_version) == 0 || len(api_version) == 0 {
+		panic("One or more environment variables not set! Exiting")
+	}
+
+	consumer := "Complete%20the%20deputy%20report"
+	account := "997462338508"
+	roletoassume := fmt.Sprintf("arn:aws:iam::%s:role/get-pact-secret-production", account)
+	pactsecretkey := "pactbroker_admin"
+
+	pact_password := getPactPassword(roletoassume, pactsecretkey)
+
+	url := "https://"  + base_url + "/pacticipants/" + consumer + "/versions/" + consumer_version
+	req, err := http.NewRequest("GET", url, nil)
+	req.Header.Add("Authorization","Basic " + basicAuth("admin",pact_password))
+
+	response, err := http.DefaultClient.Do(req)
```

Comment:
Not sure how critical it is but we're not handling this error - could be tricky to see whats gone wrong when debugging.